### DISCUSSION TITLE
[Question]:How anal to be when training images[Question]:

### DISCUSSION THREAD

**Discussion by NeoDuder** [blakeblackshear/frigate#11428] [Ask A Question]
### What is your question:

As the title really. I've been diligently going through Frigate+ images and labelling them, but part of me feels like with some of the more obscure images, I'm just training Frigate to recognise smudges as real objects. Take this image for example, it was foggy and there was a lot of blurring, but I know that those objects in the distance are cars, so have labelled them as such, marking them as "difficult" (does this have any bearing?). Is this the right thing to be doing? Should I just not be using these images at all? Even on a clear day, the cars on the other side of the street look small and smudgy, should I still be labelling them?
![Screenshot 2024-05-19 112921](https://github.com/blakeblackshear/frigate/assets/49984309/7d6a04a7-9ce6-426a-bfba-7e24dac1f57c)

**Comment by NeoDuder** [comment]
Here's a daytime example. One of those cars in the background is completely behind a bush, should I still be labelling it? Surely this is just training Frigate to recognise bushes as cars?
![image](https://github.com/blakeblackshear/frigate/assets/49984309/7a105063-3eb6-41dd-aae2-d17d8b8fd2dc)

**Comment by blakeblackshear** [comment]
Don't label things that are truly not visible. If a person looking at the image would not reasonably be able to recognize the object without prior knowledge of what is "typically" in that spot, I wouldn't label it. If the only object in the image isn't recognizable, then I often times will just delete it or not submit it.

You are using the difficult flag correctly. My plan is to use these flags to avoid training the model on them specifically, but not penalize the model during training if they are detected in the background.

**Reply by NeoDuder** [reply]
Roger that, and thanks :)
